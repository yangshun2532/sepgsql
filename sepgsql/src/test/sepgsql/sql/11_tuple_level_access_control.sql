-- 
-- testcases of tuple level access control
-- 
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: unconfined_t
-- selinux_range: s0:c0

-- SELECT, COPY
SELECT security_context, * FROM t1;
SELECT security_context, * FROM t2;

COPY t1 TO stdout;
COPY t2 TO stdout;

-- TABLE JOIN
SELECT * FROM t1 JOIN t2 ON t1.x = t2.a;

SELECT * FROM t1 LEFT OUTER JOIN t2 ON t1.x = t2.a;

SELECT * FROM t1 RIGHT OUTER JOIN t2 ON t1.x = t2.a;

SELECT * FROM t1 FULL OUTER JOIN t2 ON t1.x = t2.a;

-- UPDATE
UPDATE t1 SET y = y || '_updated' RETURNING security_context, *;
SELECT security_context, * FROM t1;

-- DELETE
DELETE FROM t2;
SELECT security_context, * FROM t2;

-- INSERT
INSERT INTO t1 VALUES (10, 'soda');

INSERT INTO t2 VALUES (10, 'orange');
