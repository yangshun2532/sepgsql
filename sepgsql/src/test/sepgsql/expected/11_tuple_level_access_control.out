-- 
-- testcases of tuple level access control
-- 
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: unconfined_t
-- selinux_range: s0:c0
-- SELECT, COPY
SELECT security_context, * FROM t1;
             security_context             | x |   y   
------------------------------------------+---+-------
 unconfined_u:object_r:sepgsql_table_t:s0 | 1 | water
 unconfined_u:object_r:sepgsql_table_t:s0 | 5 | beer
 unconfined_u:object_r:sepgsql_table_t:s0 | 6 | wine
(3 rows)

SELECT security_context, * FROM t2;
             security_context             | a |   b   
------------------------------------------+---+-------
 unconfined_u:object_r:sepgsql_table_t:s0 | 2 | blue
 unconfined_u:object_r:sepgsql_table_t:s0 | 3 | green
 unconfined_u:object_r:sepgsql_table_t:s0 | 6 | white
(3 rows)

COPY t1 TO stdout;
1	water
5	beer
6	wine
COPY t2 TO stdout;
2	blue
3	green
6	white
-- TABLE JOIN
SELECT * FROM t1 JOIN t2 ON t1.x = t2.a;
 x |  y   | a |   b   
---+------+---+-------
 6 | wine | 6 | white
(1 row)

SELECT * FROM t1 LEFT OUTER JOIN t2 ON t1.x = t2.a;
 x |   y   | a |   b   
---+-------+---+-------
 1 | water |   | 
 5 | beer  |   | 
 6 | wine  | 6 | white
(3 rows)

SELECT * FROM t1 RIGHT OUTER JOIN t2 ON t1.x = t2.a;
 x |  y   | a |   b   
---+------+---+-------
   |      | 2 | blue
   |      | 3 | green
 6 | wine | 6 | white
(3 rows)

SELECT * FROM t1 FULL OUTER JOIN t2 ON t1.x = t2.a;
 x |   y   | a |   b   
---+-------+---+-------
 1 | water |   | 
   |       | 2 | blue
   |       | 3 | green
 5 | beer  |   | 
 6 | wine  | 6 | white
(5 rows)

-- UPDATE
UPDATE t1 SET y = y || '_updated' RETURNING security_context, *;
             security_context             | x |       y       
------------------------------------------+---+---------------
 unconfined_u:object_r:sepgsql_table_t:s0 | 1 | water_updated
 unconfined_u:object_r:sepgsql_table_t:s0 | 5 | beer_updated
 unconfined_u:object_r:sepgsql_table_t:s0 | 6 | wine_updated
(3 rows)

SELECT security_context, * FROM t1;
             security_context             | x |       y       
------------------------------------------+---+---------------
 unconfined_u:object_r:sepgsql_table_t:s0 | 1 | water_updated
 unconfined_u:object_r:sepgsql_table_t:s0 | 5 | beer_updated
 unconfined_u:object_r:sepgsql_table_t:s0 | 6 | wine_updated
(3 rows)

-- DELETE
DELETE FROM t2;
SELECT security_context, * FROM t2;
 security_context | a | b 
------------------+---+---
(0 rows)

-- INSERT
INSERT INTO t1 VALUES (10, 'soda');
INSERT INTO t2 VALUES (10, 'orange');
